src/cargo/core/compiler/mod.rs

```diff
@@ -787,7 +811,7 @@ fn build_base_args<'a, 'cfg>(
     if unit.mode.is_check() {
         cmd.arg("--emit=dep-info,metadata");
     } else {
-        cmd.arg("--emit=dep-info,link");
+        cmd.arg("--emit=dep-info,metadata,link");
```

Comment:
I'm a little concerned about unconditionally emitting metadata. I can't immediately think of a problem, so maybe it doesn't matter. The only thing I can think of is that binary-only crates emit an empty file.

Reply:
Ah this is one I meant to come back and address, it's actually quite easy to only `--emit=metadata` for `BuildRmeta` targets.